src/resource/solidDataset.ts

```diff
@@ -1156,14 +1156,41 @@ function resolveLocalIrisInThing(
 }
 
 /**
- * Fetch the contents of '.well-known/solid' for a given resource URL.
+ * Fetch a SolidDataset containing information about the capabilities of the
+ * storage server that hosts the given resource URL. For more information,
+ * please see the [ESS
+ * Documentation](https://docs.inrupt.com/ess/latest/services/discovery-endpoint/#well-known-solid).
+ *
+ * **Note:** The data contained in this dataset has changed between ESS 1.1 and
+ * ESS 2.0, as such you will need to check for multiple predicates to support
+ * both versions.
+ *
+ * ```typescript
+ * const wellKnown = await getWellKnownSolid(resource);
```

Comment:
First of all: the .well-known document is broken IMHO. So no perfect solution exists.

Here are the assumptions that we can seemingly make about [the](https://docs.inrupt.com/ess/1.2/services/discovery-endpoint/#well-known-solid):
1. there will be a `solid:DiscoveryDocument`
2. it will have a `solid:notificationGateway` triple attached to it

I'm not sure if assumption 1 will always be true (not specified); but 2 would or it would blow up. So let's take it from there.

Reply:
I don't think 1 holds try for 1.1, https://docs.inrupt.com/ess/latest/services/discovery-endpoint/#well-known-solid

There it's context is `<Pod Host>/solid/v1` and I'm not even sure if this can be retrieved as text/turtle.

Reply:
Do we have a live one somewhere? Then I can verify.

Reply:
See the docs link; But regardless, I've now hidden `getWellKnownSolid` from the API docs — it's still documented in code though.